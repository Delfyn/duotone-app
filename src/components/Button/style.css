.btn {
  padding: .75rem 2.5rem;
  border-radius: 2rem;
  border: none;
  font-size: .875rem;
  font-weight: 700;
  text-transform: uppercase;
  background-color: var(--bg-dark);
  color: #fff;
  cursor: pointer;
  letter-spacing: .8px;
}

.btn:disabled {
  cursor: not-allowed;
}

.btn:hover {
  background-color: var(--primary-color);
}

.btn:hover:disabled {
  background-color: var(--bg-dark);
  cursor: not-allowed;
}
