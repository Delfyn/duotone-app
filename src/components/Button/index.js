/* @flow */

import React from 'react';
import cx from 'classnames';
import './style.css';

type ButtonProps = {
  className?: string,
  children?: Element<any>,
  onClick: Function,
  disabled?: boolean,
};

const Button = (
  { className, children, onClick, disabled = false }: ButtonProps,
) => (
  <button
    className={cx(className, 'btn')}
    onClick={onClick}
    disabled={disabled}>
    {children}
  </button>
);

export default Button;
