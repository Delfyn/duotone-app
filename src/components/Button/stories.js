/* @flow */

import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import Button from './index';

storiesOf('Button', module)
  .add('without props', () => (
    <Button onClick={action('onClick')}>Click me</Button>
  ))
  .add('disabled', () => (
    <Button disabled onClick={action('onClick')}>Click me</Button>
  ))
  .add('with onClick action', () => (
    <Button onClick={action('clicked')}>Click me</Button>
  ));
