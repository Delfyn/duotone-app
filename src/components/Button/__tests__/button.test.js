import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../index';

test('Button renders correctly', () => {
  const onClick = jest.fn();
  const component = renderer.create(
    <Button onClick={onClick}>Click here</Button>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onClick();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(onClick).toBeCalled();
});

test('Button renders className correctly', () => {
  const component = renderer.create(
    <Button className="btn-addon">Click here</Button>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
