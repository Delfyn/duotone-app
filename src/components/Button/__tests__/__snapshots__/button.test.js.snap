// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders className correctly 1`] = `
<button
  className="btn-addon btn"
  disabled={false}
  onClick={undefined}
>
  Click here
</button>
`;

exports[`Button renders correctly 1`] = `
<button
  className="btn"
  disabled={false}
  onClick={[Function]}
>
  Click here
</button>
`;

exports[`Button renders correctly 2`] = `
<button
  className="btn"
  disabled={false}
  onClick={[Function]}
>
  Click here
</button>
`;
